src/classes/RD2_ScheduleService.cls

```diff
@@ -172,6 +172,10 @@ public without sharing class RD2_ScheduleService {
         } else if (oldRd == null) {
             isNeeded = true;
 
+        // Invalid Installment Frequency should never result in a schedule update. This is a fail safe measure only
+        } else if (newRd.InstallmentFrequency__c == null || newRd.InstallmentFrequency__c < 1 || newRd.InstallmentFrequency__c > 20) {
```

Comment:
I don't understand the test for > 20 here.  Can you explain?

Reply:
It's the new validation that only allows installment between 1 and 20, everything else should be invalid.